libs/offline-tasks/src/main/java/org/opensearch/offline_tasks/TaskClient.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.offline_tasks;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.offline_tasks.task.Task;
+import org.opensearch.offline_tasks.task.TaskId;
+
+import java.util.List;
+
+/**
+ * Client used to interact with Task Store/Queue
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public interface TaskClient {
+
+    /**
+     * Submit a new task to TaskStore/Queue
+     *
+     * @param task Task to be submitted for execution on offline nodes
+     */
+    void submitTask(Task task);
+
+    /**
+     * Claim task from TaskStore/Queue. This ensures no 2 Offline Nodes work on the same task.
+     *
+     * @param taskId TaskId of the task to be claimed
+     */
+    void claimTask(TaskId taskId);
```

Comment:
Should we return a boolean to indicate success/failure of the operation?
```suggestion
    boolean claimTask(TaskId taskId);
```